docs/use_cases/retrieval_augmented_generation_eval_qdrant_ragas.md

```diff
@@ -0,0 +1,643 @@
+
+- **Evaluate Retrieval**: *context_relevancy* and *context_recall*, which give you the fair measure of the efficiency of your retrieval system.
+- **Evaluate Generation**: *faithfulness* to measure the most important aspect when working with LLM i.e. hallucinations and *answer_relevancy* which measures the relevancy of the generated answer wrt to the question.
+
+The mean score of these 4 aspects gives you the `ragas score` which is a single measure of the performance of your RAG system on the aspects of retrieval and generation.
+
+#### **RAGAS in action**
+
+To understand this further lets build a RAG system on top of Qdrant’s documentation and run the evaluations on top of it.
+We are going to be using the pre compiled [hugging-face dataset](https://www.google.com/url?q=https://huggingface.co/datasets/atitaarora/qdrant_doc&sa=D&source=editors&ust=1712248322130408&usg=AOvVaw0Bl7gKES2qQyo88tFWPaZZ) consisting of ‘text’ and ‘source’ of the documentation.
+
+![../assets/use_cases/retrieval_augmented_generation_eval_qdrant_ragas/dataset_preview.png](../assets/use_cases/retrieval_augmented_generation_eval_qdrant_ragas/dataset_preview.png)
+
+We process this dataset using the choice of our text splitter , chunks size and chunk overlap.
+
+```python
+# Retrieve the documents / dataset to be used
+dataset = load_dataset("atitaarora/qdrant_doc", split="train")
+
+#Process dataset as langchain document for further processing
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain.docstore.document import Document as LangchainDocument
+
+## Dataset to langchain document
+langchain_docs = [
+    LangchainDocument(page_content=doc["text"], metadata={"source": doc["source"]})
+    for doc in tqdm(dataset)
+]
+
+#Document chunk processing
+#Processing each document with desired TEXT_SPLITTER_ALGO , CHUNK_SIZE , CHUNK_OVERLAP etc
+
+text_splitter = RecursiveCharacterTextSplitter(
+    chunk_size=512,
+    chunk_overlap=50,
+    add_start_index=True,
+    separators=["\n\n", "\n", ".", " ", ""],
+)
+
+docs_processed = []
+for doc in langchain_docs:
+    docs_processed += text_splitter.split_documents([doc])
+```
+
+We are using [Fastembed](https://qdrant.github.io/fastembed) to make the encoding process of these document chunks lighter and simpler. FastEmbed is a lightweight, fast, Python library built for embedding generation and natively supports embedding creation suitable to use with Qdrant.
+
+```python
+## Declaring the intended Embedding Model with Fastembed
+from fastembed.embedding import TextEmbedding
+
+pd.DataFrame(TextEmbedding.list_supported_models())
+```
+
+You can execute above to get the list of all supported embedding models including the one for generating sparse vector encoding through [SPLADE++](https://huggingface.co/prithivida/Splade_PP_en_v1). We are using default model in this case.
+
+```python
```

Comment:
@atarora 
this snippet failed because the TextEmbedding class was not defined
but maybe I'm missing something?

Reply:
![Screenshot 2024-04-25 at 16 59 09](https://github.com/superlinked/VectorHub/assets/5163715/83598010-bf7b-47f6-a97b-dcea5901411d)